ov/id_pools.go

```diff
@@ -0,0 +1,338 @@
+	if len(ids) > 0 {
+		q["idList"] = ids
+	}
+
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	// Setup query
+	if len(q) > 0 {
+		c.SetQueryString(q)
+	}
+
+	data, err := c.RestAPICall(rest.GET, uri, nil)
+	if err != nil {
+		return idList, err
+	}
+
+	log.Debugf("Get Id Lists %s", data)
+	if err := json.Unmarshal(data, &idList); err != nil {
+		return idList, err
+	}
+	return idList, nil
+}
+
+// Gets a pool along with the list of ranges present in it
+func (c *OVClient) GetPoolType(poolType string) (IdPool, error) {
+	var (
+		uri    = "/rest/id-pools/"
+		idPool IdPool
+	)
+
+	if poolType == "" {
+		log.Errorf("Error submitting update request. Please provide a valid Pool Type")
+	}
+
+	uri = uri + poolType
+
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	data, err := c.RestAPICall(rest.GET, uri, nil)
+	if err != nil {
+		return idPool, err
+	}
+
+	log.Debugf("GetPoolType %s", data)
+	if err := json.Unmarshal(data, &idPool); err != nil {
+		return idPool, err
+	}
+	return idPool, nil
+}
+
+// Generates and returns a random range.
+// Used to generate a range for validation prior to actually creating it. This API is not applicable for the IPv4 and IPv6 IDs.
+func (c *OVClient) Generate(poolType string) (StartStopFragments, error) {
+	var (
+		uri = "/rest/id-pools/"
+		ids StartStopFragments
+	)
+
+	if poolType == "" {
+		log.Errorf("Error submitting update request. Please provide a valid Pool Type")
+	}
+
+	uri = uri + poolType + "/generate"
+
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	data, err := c.RestAPICall(rest.GET, uri, nil)
+	if err != nil {
+		return ids, err
+	}
+
+	log.Debugf("GenerateIds %s", data)
+	if err := json.Unmarshal(data, &ids); err != nil {
+		return ids, err
+	}
+	return ids, nil
+}
+
+// Enables or disables the pool
+func (c *OVClient) UpdatePoolType(idPool IdPool, poolType string) (IdPool, error) {
+
+	log.Infof("Initializing update of pool type for %s.", poolType)
+	var (
+		uri      = "/rest/id-pools/"
+		t        *Task
+		response IdPool
+	)
+
+	if poolType == "" {
+		log.Errorf("Error submitting update request. Please provide a valid Pool Type")
+	}
+
+	uri = uri + poolType
+
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	t = t.NewProfileTask(c)
+	t.ResetTask()
+
+	log.Debugf("REST : %s \n %+v\n", uri, idPool)
+	log.Debugf("task -> %+v", t)
+	data, err := c.RestAPICall(rest.PUT, uri, idPool)
+	if err != nil {
+		t.TaskIsDone = true
+		log.Errorf("Error submitting update request: %s", err)
+		return response, err
+	}
+
+	log.Debugf("Response update Id Pool %s", data)
+
+	dataResponse := data
+	if err := json.Unmarshal([]byte(data), &t); err != nil {
+		t.TaskIsDone = true
+		log.Errorf("Error with task un-marshal: %s", err)
+		return response, err
+	}
+	if err := json.Unmarshal([]byte(dataResponse), &response); err != nil {
```

Comment:
can you please add a comment on what is being validated in this block for my understanding.

Reply:
here we are unmarshalling the json body to array of bytes.